src/vcf/validator.cpp

```diff
@@ -122,13 +126,63 @@ namespace ebi
     }
 
     bool is_valid_vcf_file(std::istream &input,
-                           ebi::vcf::Parser &validator,
+                           const std::string &sourceName,
+                           ValidationLevel validationLevel,
+                           Ploidy ploidy,
                            std::vector<std::unique_ptr<ebi::vcf::ReportWriter>> &outputs)
     {
+        std::vector<char> line;
+        ebi::util::readline(input, line);
+        ebi::vcf::Version version;
+        try {
+            version = detect_version(line);
+        } catch (FileformatError * error) {
+            for (auto &output : outputs) {
+                output->write_error(*error);
+            }
+            return false;
+        }
+        std::unique_ptr<Parser> validator = build_parser(sourceName, validationLevel, version, ploidy);
+        return validate(line, input, *validator, outputs);
+    }
 
+    Version detect_version(const std::vector<char> &line)
```

Comment:
Maybe it's the Java programmer in me, but I understand the latter easier. I feel the iterators make the code more convoluted in this case...